templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -719,8 +719,10 @@ contents:
 
       # Make sure everything is activated
       connections=()
-      for connection in $(nmcli -g NAME c | grep -- "$MANAGED_NM_CONN_SUFFIX"); do
-        connections+=("$connection")
+      nmcli -g NAME c | while IFS= read -r connection; do
+        if [[ $connection == *"$MANAGED_NM_CONN_SUFFIX" ]]; then
+          connections+=("$connection")
+        fi
```

Comment:
`connections` is not updated

http://mywiki.wooledge.org/BashFAQ/024


```shell
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[1609]: + '[' '!' -f /etc/ovnk/extra_bridge ']'
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2262]: + nmcli connection show br-ex1
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2262]: + nmcli connection show ovs-if-phys1
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[1609]: + ovs-vsctl --timeout=30 --if-exists del-br br0
Jul 08 11:20:06 control-plane-0 ovs-vsctl[2271]: ovs|00001|vsctl|INFO|Called as ovs-vsctl --timeout=30 --if-exists del-br br0
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[1609]: + connections=()
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2272]: + nmcli -g NAME c
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ bond0 == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ens192 == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ens224 == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ bond0 == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ br-ex == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ens192 == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ens192-slave-ovs-clone == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + connections+=("$connection")
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ens224 == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ens224-slave-ovs-clone == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + connections+=("$connection")
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ovs-if-br-ex == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ovs-if-phys0 == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ovs-port-br-ex == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + [[ ovs-port-phys0 == *\-\s\l\a\v\e\-\o\v\s\-\c\l\o\n\e ]]
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + IFS=
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[2273]: + read -r connection
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[1609]: + connections+=(ovs-if-phys0 ovs-if-br-ex)
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[1609]: + '[' -f /etc/ovnk/extra_bridge ']'
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[1609]: + activate_nm_connections ovs-if-phys0 ovs-if-br-ex
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[1609]: + connections=("$@")
Jul 08 11:20:06 control-plane-0 configure-ovs.sh[1609]: + local connections

```
I set variables in a loop that's in a pipeline. Why do they disappear after the loop terminates? Or, why can't I pipe data to read?

Reply:
Process substitution I think is preferred

```shell
      while IFS= read -r connection; do
        if [[ $connection == *"$MANAGED_NM_CONN_SUFFIX" ]]; then
          connections+=("$connection")
        fi
      done < <(nmcli -g NAME c)

```